### DISCUSSION TITLE
How to do a std::system call to launch another console app?

### DISCUSSION THREAD

**Discussion by lucadesogus** [magiblot/tvision#109] [Q&A] [answered]
Hello, 

normally i use the std::system in this way to launch an executable in another console:

`std::system("start  app.exe");`

but while running tvision application it creates some troubles.
The new console is opened and closed immediately.

Can someone give some hints?

thanks

**Comment by magiblot** [comment]
Hi @lucadesogus!

What application are you trying to run with `system`? Note that console applications sometimes just print a message and exit immediately. Make sure that this isn't your case.

**Reply by lucadesogus** [reply, marked as answer]
it's not the case. 
Seems that calling the suspend() before the std::system and resume() after allows the new console to stay open.


```c++
TApplication::suspend();
std::system("start  app.exe");
TApplication::resume();
```

**Reply by magiblot** [reply]
That's indeed the only method that always works.

Although I forgot to answer earlier, I did try to reproduce the issue you were facing initially and observed that `conhost.exe` was crashing. Therefore, this misbehaviour of the console looks like a Windows bug and not something that Turbo Vision can silently work around.
